core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -18,40 +18,104 @@
 
+    /**
+     * The queries name
+     */
     public static final String NAME = "has_child";
 
-    private final QueryBuilder queryBuilder;
+    /**
+     * The default cut off point only to evaluate parent documents that contain the matching parent id terms
+     * instead of evaluating all parent docs.
+     */
+    public static final int DEFAULT_SHORT_CIRCUIT_CUTOFF = 8192;
+    /**
+     * The default maximum number of children that are required to match for the parent to be considered a match.
+     */
+    public static final int DEFAULT_MAX_CHILDREN = Integer.MAX_VALUE;
+    /**
+     * The default minimum number of children that are required to match for the parent to be considered a match.
+     */
+    public static final int DEFAULT_MIN_CHILDREN = 0;
+
+    private final QueryBuilder query;
 
-    private String childType;
+    private final String type;
 
-    private String scoreType;
+    private ScoreType scoreType = ScoreType.NONE;
 
-    private Integer minChildren;
+    private int minChildren = DEFAULT_MIN_CHILDREN;
 
-    private Integer maxChildren;
+    private int maxChildren = DEFAULT_MAX_CHILDREN;
 
-    private Integer shortCircuitCutoff;
+    private int shortCircuitCutoff = DEFAULT_SHORT_CIRCUIT_CUTOFF;
 
-    private QueryInnerHitBuilder innerHit = null;
+    private QueryInnerHits queryInnerHits;
 
-    static final HasChildQueryBuilder PROTOTYPE = new HasChildQueryBuilder(null, null);
+    static final HasChildQueryBuilder PROTOTYPE = new HasChildQueryBuilder("", EmptyQueryBuilder.PROTOTYPE);
+
+    public HasChildQueryBuilder(String type, QueryBuilder query, Integer maxChildren, Integer minChildren, Integer shortCircuitCutoff, ScoreType scoreType, QueryInnerHits queryInnerHits) {
+        this(type, query);
+        scoreType(scoreType);
+        this.scoreType = scoreType;
+        this.maxChildren = maxChildren;
+        this.minChildren = minChildren;
+        this.shortCircuitCutoff = shortCircuitCutoff;
+        this.queryInnerHits = queryInnerHits;
+    }
 
-    public HasChildQueryBuilder(String type, QueryBuilder queryBuilder) {
-        this.childType = type;
-        this.queryBuilder = queryBuilder;
+    public HasChildQueryBuilder(String type, QueryBuilder query) {
+        if (type == null) {
+            throw new IllegalArgumentException("[" + NAME + "] requires 'type' field");
+        }
+        if (query == null) {
+            throw new IllegalArgumentException("[" + NAME + "] requires 'query' field");
+        }
+        this.type = type;
+        this.query = query;
     }
 
     /**
      * Defines how the scores from the matching child documents are mapped into the parent document.
      */
-    public HasChildQueryBuilder scoreType(String scoreType) {
+    public HasChildQueryBuilder scoreType(ScoreType scoreType) {
+        if (scoreType == null) {
+            throw new IllegalArgumentException("[" + NAME + "]  requires 'score_type' field");
+        }
```

Comment:
if you want to set this to default you shouldn't pass null to this. Passing null makes no sense here at all you can just pass None and passing null is 100% of the time a bug